front-end/src/app/shared/services/report-is-editable.service.ts

```diff
@@ -0,0 +1,24 @@
+import { Injectable } from "@angular/core";
+import { Store } from "@ngrx/store";
+import { Observable, of } from "rxjs";
+import { selectActiveReport } from "../../store/active-report.selectors";
+import { Report } from "../interfaces/report.interface";
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ReportIsEditableService {
+  constructor(private store: Store){}
+
+  isEditable(): Observable<boolean> {
+    let observe = new Observable<boolean>();
+    this.store.select(selectActiveReport).subscribe((report: Report | null)=>{
+      const status = report?.upload_submission?.fec_status;
```

Comment:
`fec_status` only comes into play once it's been submitted to fec.  We don't want the report to be editable from the moment the user makes the submission.  I think this condition should take into account the `fecfile_task_state`

```
  isEditable(): Observable<boolean> {
    return this.store.select(selectActiveReport).pipe(
      map((report: Report | null) => {
        const uploadSubmission = report?.upload_submission;
        const fecStatus = report?.upload_submission?.fec_status;
        const fecfileTaskState = report?.upload_submission?.fecfile_task_state;
        return !uploadSubmission || fecStatus == 'REJECTED' || fecfileTaskState == 'FAILED';
      })
    );
  }
```

Reply:
Implemented verbatim.  Works great!